examples/core-auth-nextjs-email/src/auth.ts

```diff
@@ -0,0 +1,99 @@
+import NextAuth from 'next-auth';
+import { AuthProvider, SupportedAuthProvider } from '@toolpad/core';
+import GitHub from 'next-auth/providers/github';
+// import Credentials from 'next-auth/providers/credentials';
+
+import Nodemailer from 'next-auth/providers/nodemailer';
+import { PrismaAdapter } from '@auth/prisma-adapter';
+import type { Provider } from 'next-auth/providers';
+import { prisma } from './prisma';
+
+const providers: Provider[] = [
+  GitHub({
+    clientId: process.env.GITHUB_CLIENT_ID,
+    clientSecret: process.env.GITHUB_CLIENT_SECRET,
+  }),
+  // Credentials({
+  //   credentials: {
+  //     email: { label: 'Email Address', type: 'email' },
+  //     password: { label: 'Password', type: 'password' },
+  //   },
+  //   authorize(c) {
+  //     if (c.password !== 'password') {
+  //       return null;
+  //     }
+  //     return {
+  //       id: 'test',
+  //       name: 'Test User',
+  //       email: String(c.email),
+  //     };
+  //   },
+  // }),
+  Nodemailer({
+    server: {
+      host: process.env.EMAIL_SERVER_HOST,
+      port: process.env.EMAIL_SERVER_PORT,
+      auth: {
+        user: process.env.EMAIL_SERVER_USER,
+        pass: process.env.EMAIL_SERVER_PASSWORD,
+      },
+      secure: true,
+    },
+    from: process.env.EMAIL_FROM,
+  }),
+];
+
+export const providerMap = providers.map((provider) => {
+  if (typeof provider === 'function') {
+    const providerData = provider();
+    return {
+      id: providerData.id as SupportedAuthProvider,
+      name: providerData.name,
+    } satisfies AuthProvider;
+  }
+  return { id: provider.id as SupportedAuthProvider, name: provider.name } satisfies AuthProvider;
+});
+
+const missingVars: string[] = [];
+
+if (!process.env.GITHUB_CLIENT_ID) {
+  missingVars.push('GITHUB_CLIENT_ID');
+}
+if (!process.env.GITHUB_CLIENT_SECRET) {
+  missingVars.push('GITHUB_CLIENT_SECRET');
+}
+
+if (missingVars.length > 0) {
+  const baseMessage =
+    'Authentication is configured but the following environment variables are missing:';
+
+  if (process.env.NODE_ENV === 'production') {
+    throw new Error(`error - ${baseMessage} ${missingVars.join(', ')}`);
```

Comment:
This shouldn't throw in production and log in development. How do you feel about making this drastically simpler. Just

```tsx
if (!process.env.GITHUB_CLIENT_ID) {
  console.warn('Missing environment variable "GITHUB_CLIENT_ID"')
}
```

Just keeping it plain and simple, very readable, no extra logic, no opinions, no colors in the messages. We could let end-users make their own choices about how fancy they want this to be.

Reply:
I deployed one of the examples on a personal server to test the integration a few days ago and found the throwing in production very helpful; also personally the colours help make it more readable. Since this is personal opinion, I'm open to removing it if you think that makes more sense

Reply:
We're not (yet) in the business of configuration management. Until we are, I feel like it makes most sense to be very bare bones so that users see in the blink of an eye what this code does and how they can replace it with the configuration management of their choice.